// react-mobx/build/webpack.base.js
var config = require('../config');
var path = require('path')
const webpack = require('webpack');

function resolve (dir) {
  return path.join(__dirname,  dir)
}

module.exports = {
    entry: {
        app:"./src/index.js"
    },
    output: {
        filename: "bundle.js",
        path: __dirname + "/dist",
        publicPath: process.env.NODE_ENV === 'production'
        ? config.build.assetsPublicPath
        : config.dev.assetsPublicPath
    },
    devtool: "source-map",
    resolve: {
        extensions: [".ts", ".jsx", ".js", ".json"],
        alias: {
            'com:': resolve('/src/components'),
            'mod:': resolve('/src/modules') 
        }
    },
    module: {
        rules: [
            { test: /\.jsx?$/, 
                use:[
                    {loader: 'react-hot-loader/webpack'},
                    {loader: 'babel-loader',
                        options: {
                            presets: ['@babel/preset-env',"@babel/preset-react"],
                            plugins: [
                                ["@babel/plugin-proposal-decorators", { "legacy": true }],
                                ["@babel/plugin-proposal-class-properties", { "loose": true }],
                                '@babel/plugin-transform-runtime']
                        }
                    }
                ],
                exclude: /node_modules/,
            },
            { enforce: "pre", test: /\.js$/, loader: "source-map-loader" },
            { test:/\.css$/, loader: 'style-loader!css-loader'},
            {
                test: /\.less$/,
                use: [{
                        loader: "style-loader" 
                    }, {
                        loader: "css-loader" 
                    }, {
                        loader: "less-loader" 
                }]
            },
            {
                test: /\.(gif|jpg|png|woff|svg|eot|ttf)\??.*$/,
                loader: 'url-loader?limit=1024'
            }
        ]
    },
    plugins: [
    ]
};
